File: Projects/Project_Cinema Website/backend/src/main/java/com/cinema_website/backend/service/ShowtimeService.java
package com.cinema_website.backend.service;

import com.cinema_website.backend.dto.ShowtimeDTO;
import com.cinema_website.backend.repository.ShowtimeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@SuppressWarnings("all")
@Service
public class ShowtimeService {

    private final ShowtimeRepository showtimeRepository;

    @Autowired
    public ShowtimeService(ShowtimeRepository showtimeRepository) {
        this.showtimeRepository = showtimeRepository;
    }

    // GET
    public ShowtimeDTO getShowtime(int showtimeId) {
        return showtimeRepository.getShowtimeById(showtimeId);
    }

    public List<ShowtimeDTO> getAllShowtimes() {
        return showtimeRepository.getAllShowtimes();
    }

    public List<ShowtimeDTO> getShowtimesByMovieId(int movieId) {
        return showtimeRepository.getShowtimesByMovieId(movieId);
    }

    public List<ShowtimeDTO> getShowtimesByCinemaId(int cinemaId) {
        return showtimeRepository.getShowtimesByCinemaId(cinemaId);
    }

    public List<ShowtimeDTO> getShowtimesByHallId(int hallId) {
        return showtimeRepository.getShowtimesByHallId(hallId);
    }

    public List<ShowtimeDTO> getShowtimesByMovieDateExperience(int movieId, String showDate, String experienceType) {
        return showtimeRepository.getShowtimesByMovieDateExperience(movieId, showDate, experienceType);
    }

    public List<String> getShowtimesExperiences(int movieId, String showDate) {
        return showtimeRepository.getShowtimesExperiences(movieId, showDate);
    }



    // INSERT
    public boolean addShowtime(ShowtimeDTO showtimeDTO) {
        return showtimeRepository.addShowtime(showtimeDTO);
    }

    // UPDATE
    public boolean updateShowtime(int showtimeId, ShowtimeDTO updatedShowtime) {
        return showtimeRepository.updateShowtimeById(showtimeId, updatedShowtime);
    }

    // DELETE
    public boolean deleteShowtime(int showtimeId) {
        return showtimeRepository.deleteShowtimeById(showtimeId);
    }
}

File: Projects/Project_Cinema Website/my-react-app/src/services/ShowtimeService.js
import axios from "axios";

const baseURL = 'http://localhost:8080/api/showtimes'; // Replace with your actual API endpoint

class ShowtimeService {

    // GET
    static async getShowtimeById(showtimeId) {
        return (await axios.get(`${baseURL}/getShowtime/${showtimeId}`)).data;
    }

    static async getAllShowtimes() {
        return (await axios.get(`${baseURL}/getAllShowtimes`)).data;
    }

    static async getShowtimesByMovieId(movieId) {
        return (await axios.get(`${baseURL}/getShowtimesByMovieId/${movieId}`)).data;
    }

    static async getShowtimesByCinemaId(cinemaId) {
        return (await axios.get(`${baseURL}/getShowtimesByCinemaId/${cinemaId}`)).data;
    }

    static async getShowtimesByHallId(hallId) {
        return (await axios.get(`${baseURL}/getShowtimesByHallId/${hallId}`)).data;
    }

    static async getShowtimesByMovieDateExperience(movieId, showDate, experienceType) {
        const encodedShowDate = encodeURIComponent(showDate); // Ensure showDate is properly encoded
        return (await axios.get(`${baseURL}/getShowtimesByMovieDateExperience/${movieId}/${encodedShowDate}/${experienceType}`)).data;
    }

    static async getShowtimesExperiences(movieId, showDate) {
        const encodedShowDate = encodeURIComponent(showDate); // Ensure showDate is properly encoded
        return (await axios.get(`${baseURL}/getShowtimesExperiences/${movieId}/${encodedShowDate}`)).data;
    }

    // POST
    static async addShowtime(newShowtime) {
        return (await axios.post(`${baseURL}/addShowtime`, newShowtime)).data;
    }

    // PUT (Update)
    static async updateShowtimeById(showtimeId, updatedShowtime) {
        return (await axios.put(`${baseURL}/updateShowtime/${showtimeId}`, updatedShowtime)).data;
    }

    // DELETE
    static async deleteShowtimeById(showtimeId) {
        return (await axios.delete(`${baseURL}/deleteShowtime/${showtimeId}`)).data;
    }
}

export default ShowtimeService;
